cpp/include/rapidsmpf/memory/buffer_resource.hpp

```diff
@@ -176,6 +201,11 @@ class BufferResource {
     [[nodiscard]] MemoryReservation reserve_or_fail(size_t size, Range mem_types) {
         // try to reserve memory from the given order
         for (auto const& mem_type : mem_types) {
+            if (mem_type == MemoryType::PINNED_HOST && pinned_mr_ == nullptr) {
```

Comment:
```suggestion
            if (mem_type == MemoryType::PINNED_HOST && pinned_mr_ == PinnedMemoryResourceDisabled) {
```

Otherwise if we change the sentinel this code breaks.